### PR TITLE
Add script for setting and pushing tags

### PR DESCRIPTION
**Pull request by natemcmaster**
Per discussion on https://github.com/aspnet/Coherence-Signed/issues/771, we will set tags to match the package version release from each repo.

This generates a tag on Universe and each submodule to match the contents of version.props.

FYI - @danroth27 @muratg @Eilon

### CODE REVIEW

File: scripts/GenerateTags.ps1 [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        }
+    }
+    finally {
+        Pop-Location
+    }
+}
+
+$repoRoot = Resolve-Path "$PSScriptRoot/../"
+
+Write-Warning "Make sure you have run ``git submodule update`` first to pin the submodules to the correct"
```
Comment (ryanbrandenburg) [inline scripts/GenerateTags.ps1]: I assume the reason not to just run `git submodule update` here is that super slow on windows?

Comment (natemcmaster) [reply scripts/GenerateTags.ps1]: Correct. I could make it pause here and have you answer a yes/no question whether to proceed

File: scripts/GenerateTags.ps1 [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    param(
+        [Parameter(Mandatory = $true)]
+        [string]$Repo,
+        [Parameter(Mandatory = $true)]
+        [string]$Tag
+    )
+
+    Push-Location $Repo
+    try {
+        $tags = & git show-ref --tags
```
Comment (pranavkm) [inline scripts/GenerateTags.ps1]:
```
git show-ref --tags --verify "refs/tags/$Tag" -q
$existingTag = $?
```

File: scripts/GenerateTags.ps1 (hunk 1 above)
Comment (ryanbrandenburg) [reply scripts/GenerateTags.ps1]: Oh, I misread the what-if stuff as being a should prompt. Yeah, let's give people the chance to pause and actually do the thing here, along with a force parameter if you know what you're doing.

File: scripts/GenerateTags.ps1 [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    }
+}
+
+$repoRoot = Resolve-Path "$PSScriptRoot/../"
+
+Write-Warning "Make sure you have run ``git submodule update`` first to pin the submodules to the correct"
+
+Push-Location $repoRoot
+try {
+    [xml] $universeVersion = Get-Content "version.props"
```
Comment (pranavkm) [inline scripts/GenerateTags.ps1]: Can we add a KoreBuild target that spits out the version instead? It doesn't feel right to duplicate our version construction behavior in ps

File: scripts/GenerateTags.ps1 [hunk 4]
```diff
@@ -51,34 +51,52 @@ function New-GitTag {
     }
 }
 
+function Get-PackageVersion([string]$repoRoot) {
+    $buildScript = if (-not $IsCoreCLR -or $IsWindows) { 'build.ps1' } else { 'build.sh' }
+    $inspectTarget = "/p:CustomAfterKoreBuildTargets=$PSScriptRoot/GetPackageVersion.targets"
```
Comment (pranavkm) [inline scripts/GenerateTags.ps1]: Nice!

**Comment by natemcmaster** [comment]
:up: :date: to include feedback.

On the dev branch, this is what the script looks like now
```
$ ./scripts/GenerateTags.ps1 -whatif
WARNING: Make sure you have run `git submodule update` first to pin the submodules to the correct commit

This will apply tags to all submodules
Continue?
[Y] Yes  [N] No  [S] Suspend  [?] Help (default is "Y"): Y
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/Antiforgery".
WARNING: AuthSamples does not have a version.props file. Skipping
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/AzureIntegration".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/BasicMiddleware".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/BrowserLink".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/Caching".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/Common".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/Configuration".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/CORS".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/DataProtection".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/DependencyInjection".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/Diagnostics".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/DotNetTools".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/EntityFrameworkCore".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/EventNotification".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/FileSystem".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/Hosting".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/HtmlAbstractions".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/HttpAbstractions".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/HttpClientFactory".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/HttpSysServer".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/Identity".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/IISIntegration".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/JavaScriptServices".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/JsonPatch".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/KestrelHttpServer".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/Localization".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/Logging".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/MetaPackages".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/Microsoft.Data.Sqlite".
WARNING: MusicStore does not have a version.props file. Skipping
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/Mvc".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/MvcPrecompilation".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/Options".
WARNING: Proxy: version (0.3.0-preview2-final) does not match universe (2.1.0-preview2-final)
What if: Performing the operation "Tag 0.3.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/Proxy".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/Razor".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/ResponseCaching".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/Routing".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/Scaffolding".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/Security".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/ServerTests".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/Session".
WARNING: SignalR: version (1.0.0-preview2-final) does not match universe (2.1.0-preview2-final)
What if: Performing the operation "Tag 1.0.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/SignalR".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/StaticFiles".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/Templating".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/Testing".
WARNING: WebHooks: version (1.0.0-preview2-final) does not match universe (2.1.0-preview2-final)
What if: Performing the operation "Tag 1.0.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/WebHooks".
What if: Performing the operation "Tag 2.1.0-preview2-final" on target "/Users/namc/dev/aspnet/Universe/modules/WebSockets".
```

**Pull request merged**